### DISCUSSION TITLE
MaximumDepthExceeded Error throwing up  from `notifyManager` of tanstack/query package

### DISCUSSION THREAD

**Discussion by sagarchoudhary96** [TanStack/query#6550] [Q&A]
i have a navigation component rendering lots of navItems and each of them are having some useQuery() calls when are done to get some count for each navItem if enabled . even though i'm enabling the queries only for navItem for which it should be fetched and passing `enabled: false` as option to useQuery and useQueries() where not needed to be fetched, i get below error which is crashing the application on mobile. I'm unable to figure out if something can be fixed from our end for it or it's something to be fixed in this `query` package itself 

<img width="406" alt="Screenshot 2023-12-18 at 4 40 29 PM" src="https://github.com/TanStack/query/assets/16102594/4955e7fc-24d9-4158-a26f-a9ac38ea98d5">

**Comment by sagarchoudhary96** [comment]
any help here ? 

@tannerlinsley

**Reply by TkDodo** [reply]
I would need to see this in a codesandbox please
